lib/types/inet-address.js

```diff
@@ -10,238 +11,243 @@ const utils = require("../utils");
  * @param {Buffer} buffer
  * @constructor
  */
-function InetAddress(buffer) {
-    if (
-        !(buffer instanceof Buffer) ||
-        (buffer.length !== 4 && buffer.length !== 16)
-    ) {
-        throw new TypeError("The ip address must contain 4 or 16 bytes");
+class InetAddress {
+    constructor(buffer) {
+        this._core = rust.InetAddressWrapper.new(buffer);
     }
 
-    /**
-     * Immutable buffer that represents the IP address
-     * @type Array
-     */
-    this.buffer = buffer;
-
     /**
      * Returns the length of the underlying buffer
```

Comment:
Maybe explicitly say it's read only?
From what I understand there is nothing stoping in the old driver from editing internal fields (though it may generate UB?).

Reply:
From Datastax API:
> buffer: Immutable buffer that represents the IP address
So by exposing only getter I assure immutability
But I will add @readonly to readonly properties